### ISSUE TITLE
CA2000 Use recommended dispose pattern to ensure that object created

### ISSUE THREAD

**Issue by YuliiaKovalova** [dotnet/roslyn-analyzers#7293]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

**Steps to Reproduce**:
This code in MSBuild generates CA2000:
https://github.com/dotnet/msbuild/blob/6fec6072f0591bede57a71474ddc08b95e3e4c28/src/Build/Evaluation/IntrinsicFunctions.cs#L294

**Expected Behavior**:
No warning occurs.

**Actual Behavior**:
The warning is displayed.

**Comment by symbiogenesis** [comment]
Consider, also:

```csharp
var asyncDisposable = new AsyncDisposable();

await using (asyncDisposable.ConfigureAwait(false))
{
    Foo();
}
```

I would not expect this to result in a warning as long as `asyncDisposable` is not accessed again outside of the `await using` block.

**Note:** In the event of `ConfigureAwait()` being applied, the disposable becomes type `ConfiguredAsyncDisposable`
